tests/integration/initramfs.bats

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bats
+
+load helpers
+
+# Rather than building our own kernel for use with qemu, just reuse the host's
+# kernel since we just need some kernel that supports containers that we can
+# use to run our custom initramfs.
+function find_vmlinuz() {
+	shopt -s nullglob
+	candidates=(
+		/boot/vmlinuz
+		/boot/vmlinuz-"$(uname -r)"*
+		/usr/lib*/modules/"$(uname -r)"/vmlinu* # vmlinuz / vmlinux.{gz,xz,*}
+	)
+	shopt -u nullglob
+
+	for candidate in "${candidates[@]}"; do
+		[ -e "$candidate" ] || continue
+		export HOST_KERNEL="$candidate"
+		return 0
+	done
+
+	# Actuated doesn't provide a copy of the boot kernel, so we have to skip
+	# the test in that case. It also seems they don't allow aarch64 guests
+	# either (see <https://docs.actuated.com/examples/kvm-guest/>).
+	skip "could not find host vmlinuz kernel"
+}
+
+function setup() {
+	INITRAMFS_ROOT="$(mktemp -d "$BATS_RUN_TMPDIR/runc-initramfs.XXXXXX")"
+	find_vmlinuz
+}
+
+function teardown() {
+	[ -v INITRAMFS_ROOT ] && rm -rf "$INITRAMFS_ROOT"
+}
+
+function qemu_native() {
+	# Different distributions put qemu-kvm in different locations and with
+	# different names. Debian and Ubuntu have a "kvm" binary, while AlmaLinux
+	# has /usr/libexec/qemu-kvm.
+	qemu_candidates=("kvm" "qemu-kvm" "/usr/libexec/qemu-kvm")
+
+	qemu_binary=
+	for candidate in "${qemu_candidates[@]}"; do
+		"$candidate" -help &>/dev/null || continue
+		qemu_binary="$candidate"
+		break
+	done
+	# TODO: Maybe we should also try to call qemu-system-FOO for the current
+	# architecture if qemu-kvm is missing?
+	[ -n "$qemu_binary" ] || skip "could not find qemu-kvm binary"
+
+	case "$(go env GOARCH)" in
+	386 | amd64)
+		machine="pc"
+		;;
+	arm | arm64)
+		machine="virt"
+		;;
+	*)
+		echo "could not figure out -machine argument for qemu -- using default" >&2
+		;;
+	esac
+	machine_args=()
+	[ -n "$machine" ] && machine_args=("-machine" "$machine")
+
+	qemu_output="$INITRAMFS_ROOT/qemu.log"
+	sane_run --pipe --timeout=3m \
+		"$qemu_binary" "${machine_args[@]}" "$@" \| tee -a "$qemu_output"
```

Comment:
So we have `tee`  saving qemu output to a file, and then `bats` collecting output to `$output` env var. Seems like an unnecessary complication, or am I missing something?

Frankly, you can just do the same thing without bats, emulating `run` in place. Something like
```bash
local log="$INITRAMFS_ROOT/qemu.log"
status=0
timeout --foreground --signal=KILL 3m \
  "$qemu_binary" "${machine_args[@]}" "$@" > $log 2>&1 || status=$?
output="$(cat $log)"
```

The only difference, I guess, is `$lines` array is not populated.

Reply:
This bit is me trying to debug why almalinux 9 is failing and it seems that the output log in Cirrus is missing some lines so I'm trying to figure out why the kernel is panicking.

Reply:
Okay, I debugged the issue -- you needed to use `-cpu max` (or `-cpu host` but that's an x86-only thing) so that the host glibc can run inside the qemu guest on AlmaLinux 9.